java-checks/src/test/files/checks/MethodParametersOrderCheck.java

```diff
@@ -0,0 +1,49 @@
+class A{
+  
+  public double divide(int divisor, int dividend) {
+    return divisor/dividend;
+  }
+
+  public void doTheThing() {
+    int divisor = 15;
+    int dividend = 5;
+
+    double result = divide(dividend, divisor);  // Noncompliant [[sc=21;ec=46]] {{Parameters to divide have the same names but not the same order as the method arguments.}}
+  }
+  
+  public void doTheThing() {
+    int divisor = 15;
+    int dividend = 5;
+    double result = divide(divisor, dividend);  // Compliant
+  }
+  
+  int fun();
+  
+  void test(int parameter1, int parameter2) {
+    int divisOr = 15;
+    int dividend = 5;
+  
+    int result = divide(dividend, divisOr);  // Noncompliant {{Parameters to divide have the same names but not the same order as the method arguments.}}
+  
+    result = divide(divisOr, dividend);      // Compliant
+    result = divide(parameter1, parameter2);  // Compliant
+    result = divide(fun(), dividend);         //Compliant
+  
+    // coverage
+    fun();
```

Comment:
Ok, so you tested with zero, and two parameters... What about 1 or 3 ? What about variadic methods ? Don't focus on special cases.

```
void foo(int a, int ... bs) { /* ... */ }
```